### PR TITLE
Refactor Authpipe

### LINKED ISSUE #451: Refactoring Backend
This should have a brief overview of all parts of refactoring the frontend. Do not be specific in this issue; specifics should be described in their own separate issue.

Issues listed by priority (subject to change):
* #452 
* #453
* #455
* #454
* #461

### LINKED ISSUE #455: Add Backend Style Guide
This should be an ongoing process; we want to ensure that our developers follow the same programming style (which may also be enforced by linters and auto-formatters). This is to ensure readability and keep a consistent codebase.

### PR DESCRIPTION
**Pull request by JJamesWWang**
### Description
Refactoring for authpipe django app
### Tasks
- [x] Add tests if necessary
- [x] Refactor code

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,8 +25,6 @@ COPY ./build/local_settings.py /code/semesterly/local_settings.py
 COPY ./build/run_parser.sh /code/run_parser.sh
 
 RUN pip3 install -r /code/requirements.txt
-# This is needed on newer ubuntu
-RUN pip3 install psycopg2-binary==2.8.6
```
Comment (yesterd3v) [inline Dockerfile:29]: Did you manually validate locally that this doesn't break the build?

File: docs/styleguide.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.. _styleguide:
```
Comment (yesterd3v) [inline docs/styleguide.rst:1]: Note that to update the docs, you'll have to manually rebuild the docs via the readthedocs portal.
It hasn't yet been officially connected to this repository to auto-build.

Comment (JJamesWWang) [reply docs/styleguide.rst:1]: That's okay, it's not really complete yet anyways

File: Dockerfile (hunk 1 above)
Comment (JJamesWWang) [reply Dockerfile:29]: Yes, this doesn't break it since I added it to the requirements.txt

**Pull request merged**